gwctl/pkg/printer/backends_test.go

```diff
@@ -225,3 +225,281 @@ ns1        foo-svc-2  Service  2d   ns1/foo-httproute-2, ns1/foo-httproute-3 + 2
+				},
+			},
+		},
+	}
+
+	objects := []runtime.Object{
+		&gatewayv1.GatewayClass{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: gatewayv1.GroupVersion.String(),
+				Kind:       "GatewayClass",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo-gatewayclass",
+			},
+			Spec: gatewayv1.GatewayClassSpec{
+				ControllerName: "example.net/gateway-controller",
+				Description:    common.PtrTo("random"),
+			},
+		},
+
+		common.NamespaceForTest("default"),
+		&gatewayv1.Gateway{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: gatewayv1.GroupVersion.String(),
+				Kind:       "Gateway",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-gateway",
+				Namespace: "default",
+			},
+			Spec: gatewayv1.GatewaySpec{
+				GatewayClassName: "foo-gatewayclass",
+			},
+		},
+		&unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "foo.com/v1",
+				"kind":       "HealthCheckPolicy",
+				"metadata": map[string]interface{}{
+					"name":      "health-check-gateway",
+					"namespace": "default",
+				},
+				"spec": map[string]interface{}{
+					"override": map[string]interface{}{
+						"key1": "value-child-1",
+					},
+					"default": map[string]interface{}{
+						"key2": "value-child-2",
+						"key5": "value-child-5",
+					},
+					"targetRef": map[string]interface{}{
+						"group":     "gateway.networking.k8s.io",
+						"kind":      "Gateway",
+						"name":      "foo-gateway",
+						"namespace": "default",
+					},
+				},
+			},
+		},
+
+		&gatewayv1.HTTPRoute{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: gatewayv1.GroupVersion.String(),
+				Kind:       "HTTPRoute",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-httproute",
+				Namespace: "default",
+			},
+			Spec: gatewayv1.HTTPRouteSpec{
+				CommonRouteSpec: gatewayv1.CommonRouteSpec{
+					ParentRefs: []gatewayv1.ParentReference{{
+						Kind:  common.PtrTo(gatewayv1.Kind("Gateway")),
+						Group: common.PtrTo(gatewayv1.Group("gateway.networking.k8s.io")),
+						Name:  "foo-gateway",
+					}},
+				},
+				Rules: []gatewayv1.HTTPRouteRule{
+					{
+						BackendRefs: []gatewayv1.HTTPBackendRef{
+							{
+								BackendRef: gatewayv1.BackendRef{
+									BackendObjectReference: gatewayv1.BackendObjectReference{
+										Port: ptr.To(gatewayv1.PortNumber(8080)),
+										Name: gatewayv1.ObjectName("foo-svc"),
+										Kind: ptr.To(gatewayv1.Kind("Service")),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		&unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "foo.com/v1",
+				"kind":       "HealthCheckPolicy",
+				"metadata": map[string]interface{}{
+					"name":              "health-check-gatewayclass",
+					"namespace":         "default",
+					"creationTimestamp": fakeClock.Now().Add(-6 * 24 * time.Hour).Format(time.RFC3339),
+				},
+				"spec": map[string]interface{}{
+					"default": map[string]interface{}{
+						"key2": "value-parent-2",
```

Comment:
Worth adding an example of a default `"key1": "value-parent-1"` here perhaps, to highlight the difference between an override vs a default inherited from the gateway?

```suggestion
						"key1": "value-parent-1",
						"key2": "value-parent-2",
```

Reply:
Thanks! Done.